packages/datadog-instrumentations/src/mongodb-core.js

```diff
@@ -0,0 +1,188 @@
+'use strict'
+
+const {
+  channel,
+  addHook,
+  AsyncResource
+} = require('./helpers/instrument')
+const shimmer = require('../../datadog-shimmer')
+
+global.foo = channel('apm:mongodb:insert:start')
+
+addHook({ name: 'mongodb-core', versions: ['2 - 3.1.9'] }, Server => {
+  Server.Server.prototype.command = wrapCommand(Server.Server.prototype.command, 'command')
+  Server.Server.prototype.insert = wrapCommand(Server.Server.prototype.insert, 'insert', 'insert')
+  Server.Server.prototype.update = wrapCommand(Server.Server.prototype.update, 'update', 'update')
+  Server.Server.prototype.remove = wrapCommand(Server.Server.prototype.remove, 'remove', 'remove')
+
+  Server.Cursor.prototype._getmore = wrapCursor(Server.Cursor.prototype._getmore, 'getMore', 'getMore')
+  Server.Cursor.prototype._find = wrapQuery(Server.Cursor.prototype._find, '_find')
+  Server.Cursor.prototype.kill = wrapCursor(Server.Cursor.prototype.kill, 'killCursors', 'killCursors')
+  return Server
+})
+
+addHook({ name: 'mongodb', versions: ['>=4'], file: 'lib/cmap/connection.js' }, Connection => {
+  const proto = Connection.Connection.prototype
+  proto.command = wrapConnectionCommand(proto.command, 'command')
+  proto.query = wrapConnectionCommand(proto.query, 'query')
+  proto.getMore = wrapConnectionCommand(proto.getMore, 'getMore', 'getMore')
+  proto.killCursors = wrapConnectionCommand(proto.killCursors, 'getMore', 'getMore')
+  return Connection
+})
+
+addHook({ name: 'mongodb', versions: ['>=3.3 <4'], file: 'lib/core/wireprotocol/index.js' }, wp => {
+  patch(wp)
+  return wp
+})
+
+addHook({ name: 'mongodb-core', versions: ['>=3.2'], file: 'lib/wireprotocol/index.js' }, wp => {
+  patch(wp)
+  return wp
+})
+
+addHook({ name: 'mongodb-core', versions: ['~3.1.10'], file: 'lib/wireprotocol/3_2_support.js' }, WireProtocol => {
+  WireProtocol.prototype.command = wrapUnifiedCommand(WireProtocol.prototype.command, 'command')
```

Comment:
You should use `shimmer.wrap()` in here, like `shimmer.wrap(WireProtocol.prototype, 'command', .......)`.

Also since this and the next one are the same code, you can do it in one function passed as the callback in both cases.